functions/utils/cache.ts

```diff
@@ -0,0 +1,26 @@
+interface Data {
+  title: string
+  image: string
+  url: string
+}
+
+class Cache {
+  async match(request: string): Promise<Data | undefined> {
+    const cache = await caches.open('function-cache')
+    const response = await cache.match(request)
+    if (!response) return undefined
+    const data: Data = JSON.parse(await response.text())
+    if (!data.title || !data.image || !data.url) throw new Error('Invalid data')
```

Comment:
Should we be throwing or returning undefined so that it can be re-fetched?

Reply:
This could also be a test case, to ensure that it continue to do what we expect (either throw or return undefined).

Reply:
returning undefined makes more sense yeah!